### PR TITLE
Editor add obstacles

### PR DESCRIPTION
**Pull request by hellochar**
Adds obstacle toJSON/fromJSON, rendering, and adding in editor.

left-click drag in editor to add obstacles.

@strayhobo

### CODE REVIEW

File: lib/static/lib/js/ui_renderer.js [hunk 1]
```diff
@@ -77,6 +77,16 @@ UIRenderer.prototype.renderUnitSelectedHighlight = function(unit) {
     this.drawCircle(unit.pos.x, unit.pos.y, unit.size - Math.floor(this.ctx.lineWidth / 2), '#00CC00');
 };
 
+UIRenderer.prototype.renderConnection = function(startNode, endNode) {
```
Comment (jamesmuerle) [inline lib/static/lib/js/ui_renderer.js]: Call this method renderNodeConnection

File: lib/static/lib/js/ui_renderer.js [hunk 2]
```diff
@@ -144,6 +154,12 @@ UIRenderer.prototype.renderGS = function(gamestate) {
         }.bind(this));
     }.bind(this));
 
+    gamestate.obstacles.nodes.forEach(function (startNode) {
+        startNode.connections.forEach(function (endNode) {
+            this.renderConnection(startNode, endNode);
+        }.bind(this));
+    }.bind(this));
+
```
Comment (jamesmuerle) [inline lib/static/lib/js/ui_renderer.js:162]: Drawing each wall twice. Just keep it in mind.

File: jasmine_testing/static/jasmine_testing/specs/GameStateSpec.js [hunk 3]
```diff
@@ -69,6 +70,33 @@ describe("GameState", function() {
             });
         });
 
+        describe("addWall", function() {
+
+            // beforeEach(function () {
```
Comment (kfang) [inline jasmine_testing/static/jasmine_testing/specs/GameStateSpec.js:78]: why is this here.... comment pls!

**Comment by jamesmuerle** [comment]
OH MY GAWD YOU DID IT!
